app/services/statements/search.rb

```diff
@@ -0,0 +1,65 @@
+module Statements
+  class Search
+    class InvalidFeeTypeError < StandardError; end
+
+    include Queries::FilterIgnorable
+    include Queries::ConditionFormats
+
+    attr_reader :scope
+
+    def initialize(lead_provider_id: :ignore, contract_period_years: :ignore, fee_type: 'output', statement_date: :ignore, order_by: :payment_date)
+      @scope = Statement.distinct.includes(active_lead_provider: %i[lead_provider contract_period])
+
+      where_lead_provider_is(lead_provider_id)
+      where_contract_period_year_in(contract_period_years)
+      where_fee_type_is(fee_type)
+      where_statement_date(statement_date)
+      set_order_by(order_by)
+    end
+
+    def statements
+      scope
+    end
+
+  private
+
+    def where_lead_provider_is(lead_provider_id)
+      return if ignore?(filter: lead_provider_id)
+
+      scope.merge!(Statement.joins(:lead_provider).where(lead_providers: { id: lead_provider_id }))
+    end
+
+    def where_contract_period_year_in(contract_period_years)
+      return if ignore?(filter: contract_period_years)
+
+      scope.merge!(Statement.joins(:contract_period).where(contract_periods: { year: extract_conditions(contract_period_years) }))
```

Comment:
I don't feel that strongly about it, but I find `merge!` less natural than just re-assigning the scope:

```
@scope = scope.joins(:contract_period).where(contract_periods: { year: extract_conditions(contract_period_years) }
```

I may have changed all these in the API queries on my other PR 🙄

Reply:
updated :)